randovania/gui/tracker_window.py

```diff
@@ -435,6 +437,20 @@ def update_locations_tree_for_reachable_nodes(self):
                         is_visible = False
 
                     node_item.setHidden(not is_visible)
+                    node_text = self._item_to_node[node_item].name
+                    # Check if you can reach back to current node from that node
+                    if is_visible:
```

Comment:
calculating safe nodes is awfully expensive but also *surely* there should be a dedicated function for it rather than duplicating the logic like this?